File: Facility Management_Decompiler/Facility Management/mCheckSum.cs
using Facility_Management.My;
using System;
using System.IO;
using System.Runtime.CompilerServices;
using System.Text;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace Facility_Management
{
	internal sealed class mCheckSum
	{
		private static readonly int[] Crc32Table = new int[256];

		private static int InitCrc32(int Seed = -306674912, int Precondition = -1)
		{
			short num = 0;
			checked
			{
				do
				{
					int num2 = num;
					short num3 = 0;
					do
					{
						int num4 = num2 & -2;
						num4 = unchecked(num4 / 2);
						num4 &= 0x7FFFFFFF;
						num2 = (((num2 & 1) == 0) ? num4 : (num4 ^ Seed));
						num3 = (short)unchecked(num3 + 1);
					}
					while (num3 <= 7);
					Crc32Table[num] = num2;
					num = (short)unchecked(num + 1);
				}
				while (num <= 255);
				return Precondition;
			}
		}

		private static int AddCrc32(ref byte[] Data, int CRC32)
		{
			int num = Information.UBound(Data);
			for (int i = 0; i <= num; i = checked(i + 1))
			{
				byte b = Data[i];
				int num2 = ((CRC32 & -256) / 256) & 0xFFFFFF;
				int num3 = CRC32 & 0xFF;
				num3 ^= b;
				int num4 = Crc32Table[num3];
				CRC32 = num2 ^ num4;
			}
			return CRC32;
		}

		private static int GetCrc32(int CRC32)
		{
			return CRC32 ^ -1;
		}

		[MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
		public static string Append(string szFileName, bool CreateCRC = true)
		{
			if (Strings.StrComp(mFunction.ParsePath(Application.ExecutablePath, 4), mFunction.ParsePath(szFileName, 4), CompareMethod.Text) == 0)
			{
				CreateCRC = false;
			}
			string path = mFunction.ParsePath(szFileName, 7) + MyProject.Forms.FormBox.mExtFileCRC;
			if (File.Exists(path))
			{
				File.Delete(path);
			}
			string text = szFileName + MyProject.Forms.FormBox.mExtFileCRC;
			string text2;
			if (File.Exists(text) && CreateCRC)
			{
				if (DateTime.Compare(File.GetLastWriteTime(text), File.GetLastWriteTime(szFileName)) == 0)
				{
					text2 = File.ReadAllText(text, Encoding.Default);
				}
				else
				{
					if (Interaction.MsgBox("File date error: " + text + "\r\n\r\nDo you want to create it again?", MsgBoxStyle.Exclamation | MsgBoxStyle.YesNo, "CRC error") != MsgBoxResult.No)
					{
						File.Delete(text);
						goto IL_00d1;
					}
					text2 = File.ReadAllText(text, Encoding.Default);
				}
				goto IL_026d;
			}
			if (File.Exists(text))
			{
				File.Delete(text);
			}
			goto IL_00d1;
		IL_026d:
			return text2;
		IL_00d1:
			int num = 0;
			int num2 = 1048576;
			text2 = "";
			MyProject.Forms.FormBox.ToolStripStatusLabelFM.Text = "Check CRC: " + mFunction.ParsePath(szFileName, 5);
			checked
			{
				short num3 = (short)FileSystem.FreeFile();
				FileSystem.FileOpen(num3, szFileName, OpenMode.Binary, OpenAccess.Read);
				int num4 = (int)FileSystem.LOF(num3);
				Application.DoEvents();
				if (num4 <= 0)
				{
					FileSystem.FileClose(num3);
				}
				else
				{
					int cRC = InitCrc32();
					while (num < num4)
					{
						int num5 = (int)Math.Round((double)num / (double)num4 * 100.0);
						if (num5 != MyProject.Forms.FormBox.ToolStripProgressBar1.Value)
						{
							MyProject.Forms.FormBox.ToolStripProgressBar1.Value = num5;
							Application.DoEvents();
						}
						if (num4 - num < num2)
						{
							num2 = num4 - num;
						}
						byte[] array = new byte[num2 - 1 + 1];
						Array Value = array;
						FileSystem.FileGet(num3, ref Value, -1L);
						array = (byte[])Value;
						num += num2;
						cRC = AddCrc32(ref array, cRC);
					}
					FileSystem.FileClose(num3);
					MyProject.Forms.FormBox.ToolStripProgressBar1.Value = 100;
					cRC = GetCrc32(cRC);
					text2 = Conversion.Hex(cRC);
					if (CreateCRC)
					{
						File.WriteAllText(text, text2, Encoding.Default);
						File.SetCreationTime(text, File.GetCreationTime(szFileName));
						File.SetLastAccessTime(text, File.GetLastAccessTime(szFileName));
						File.SetLastWriteTime(text, File.GetLastWriteTime(szFileName));
					}
					MyProject.Forms.FormBox.ToolStripStatusLabelFM.Text = null;
				}
				goto IL_026d;
			}
		}

		public static string AppendString(string svData)
		{
			int i = 1;
			int num = Strings.Len(svData);
			checked
			{
				byte[] Data = new byte[num - 1 + 1];
				for (; i < num; i += 2)
				{
					Application.DoEvents();
					Data[i - 1] = (byte)Strings.Asc(Strings.Mid(svData, i, 2));
				}
				int cRC = InitCrc32();
				cRC = AddCrc32(ref Data, cRC);
				cRC = GetCrc32(cRC);
				return Conversion.Hex(cRC);
			}
		}
	}
}

File: Récupération 20210215/Facility Management 1.4.1.7/Module/mCheckSum.vb

Option Strict Off
Option Explicit On

Module mCheckSum

    ' Ajout de DoEvents

    '|\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
    '|     CRC32 Check/Append      \
    '| Tutorial by E.B.ArtSoft@     \
    '| email : EBArtSoft@aol.com     \
    '| Source: Original CRC32 module  \
    '|         coded by Neo            \
    '|         http://vbcode.8m.com/    \
    '|\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\

    Private ReadOnly Crc32Table(255) As Integer

    Private Function InitCrc32(Optional ByVal Seed As Integer = &HEDB88320, Optional ByVal Precondition As Integer = &HFFFFFFFF) As Integer
        'Declaration des variables
        Dim lTempCrc32 As Integer
        Dim iBytes As Short
        Dim iBits As Short
        Dim lCrc32 As Integer
        'Boucle commence a 0 fini a 255 soit 256 cycles
        For iBytes = 0 To 255
            'CRC temporaire = compteur de cycle
            lCrc32 = iBytes
            'Effectue un decalage de bit vers la droite
            For iBits = 0 To 7
                'Masque le bits de poid faible
                lTempCrc32 = lCrc32 And &HFFFFFFFE
                'Decale les bit vers la droite
                lTempCrc32 \= &H2
                'Masque le bit de poid fort
                lTempCrc32 = lTempCrc32 And &H7FFFFFFF
                'Verifie si le CRC = 0 et mix avec la valeur Seed
                If (lCrc32 And &H1) <> 0 Then
                    lCrc32 = lTempCrc32 Xor Seed
                Else
                    lCrc32 = lTempCrc32
                End If
            Next
            'Ajoute le CRC dans la table
            Crc32Table(iBytes) = lCrc32
        Next
        'Renvoi la precondition
        InitCrc32 = Precondition
    End Function

    Private Function AddCrc32(ByRef Data() As Byte, ByVal CRC32 As Integer) As Integer
        'Declaration des variables
        Dim lTableValue As Integer
        Dim iCounter As Integer
        Dim bCharValue As Byte
        Dim lAccValue As Integer
        Dim lIndex As Integer

        Dim IndexMax As Integer = UBound(Data)

        'On vas verifier toute les données du fichier
        'Donc boucle de 0 à [Fin du fichier]
        For iCounter = 0 To IndexMax

            'Var temporaire = donnée du fichier
            'à la position icounter
            bCharValue = Data(iCounter)
            'Masque les bits 0 à 255 (poid faible)
            lAccValue = CRC32 And &HFFFFFF00
            'Divise par 256 (decalage vers la droite)
            lAccValue \= &H100
            'Masque les bits de poid fort
            lAccValue = lAccValue And &HFFFFFF
            'Recupere le module de 256
            lIndex = CRC32 And &HFF
            'Combine avec notre var temp
            lIndex = lIndex Xor bCharValue
            'Recupere le CRC dans notre table
            'Selon l'index donné par notre calcule
            lTableValue = Crc32Table(lIndex)
            'Puis on combine avec notre valeur
            CRC32 = lAccValue Xor lTableValue

        Next
        'Renvoi le CRC
        AddCrc32 = CRC32

    End Function

    Private Function GetCrc32(ByVal CRC32 As Integer) As Integer
        'Renvoi le ou exclusif du CRC
        GetCrc32 = CRC32 Xor &HFFFFFFFF
    End Function

    Public Function Append(ByVal szFileName As String, Optional ByVal CreateCRC As Boolean = True) As String

        If StrComp(ParsePath(Application.ExecutablePath, FILENAME), ParsePath(szFileName, FILENAME), CompareMethod.Text) = 0 Then
            CreateCRC = False
        End If

        Dim svFileOldCRC As String = ParsePath(szFileName, PATH_FILENAME_ONLY) + FormBox.mExtFileCRC
        If IO.File.Exists(svFileOldCRC) Then
            IO.File.Delete(svFileOldCRC)
        End If
        Dim svFileCRC As String = szFileName + FormBox.mExtFileCRC

        If IO.File.Exists(svFileCRC) And CreateCRC Then

            If DateTime.Compare(IO.File.GetLastWriteTime(svFileCRC), IO.File.GetLastWriteTime(szFileName)) = 0 Then
                Return (IO.File.ReadAllText(svFileCRC, System.Text.Encoding.Default))

            Else
                If MsgBox("File date error: " + svFileCRC + vbNewLine + vbNewLine + "Do you want to create it again?", CType(MsgBoxStyle.Exclamation + MsgBoxStyle.YesNo + MsgBoxStyle.DefaultButton1, MsgBoxStyle), "CRC error") = MsgBoxResult.No Then
                    Return (IO.File.ReadAllText(svFileCRC, System.Text.Encoding.Default))
                Else
                    IO.File.Delete(svFileCRC)
                End If
            End If

        ElseIf IO.File.Exists(svFileCRC) Then
            IO.File.Delete(svFileCRC)
        End If


        'Declaration des variables
        Dim lCrc32Value As Integer
        Dim nCount As Integer = 0
        Dim data() As Byte
        Dim buffer As Integer = 1024 * 1024

        'Si erreur on vas au label "Fin"
        'On Error GoTo Fin

        Append = ""

        'Ajout ProgressBar CRC
        FormBox.ToolStripStatusLabelFM.Text = "Check CRC: " + ParsePath(szFileName, FILENAME_ONLY)

        'Ouvre le fichier
        Dim FileNumber As Short = CShort(FreeFile())
        FileOpen(FileNumber, szFileName, OpenMode.Binary, OpenAccess.Read)
        'Recupere la taille du fichier
        Dim FileSize As Integer = CInt(LOF(FileNumber))


        System.Windows.Forms.Application.DoEvents()

        'Seulement si le fichier n'est pas vide
        If FileSize <= 0 Then
            'Fin de la procedure
            'Le fchier est vide
            FileClose(FileNumber)
            Exit Function
        End If

        'Initialise le tableau du CRC
        lCrc32Value = InitCrc32()

        'Calcule le CRC
        While nCount < FileSize

            Dim nValue As Integer = CInt((nCount / FileSize) * 100)

            If nValue <> FormBox.ToolStripProgressBar1.Value Then
                FormBox.ToolStripProgressBar1.Value = nValue
                System.Windows.Forms.Application.DoEvents()
            End If

            If FileSize - nCount < buffer Then
                buffer = FileSize - nCount
            End If

            'Charge les données du fichier dans le buffer
            ReDim data(buffer - 1)
            FileGet(FileNumber, data)

            nCount += buffer

            lCrc32Value = AddCrc32(data, lCrc32Value)

        End While

        FileClose(FileNumber)

        FormBox.ToolStripProgressBar1.Value = 100

        'Ajouste le CRC par XOR
        lCrc32Value = GetCrc32(lCrc32Value)

        'Info
        Append = CStr(Hex(lCrc32Value))

        If CreateCRC Then

            IO.File.WriteAllText(svFileCRC, Append, System.Text.Encoding.Default)

            IO.File.SetCreationTime(svFileCRC, IO.File.GetCreationTime(szFileName))
            IO.File.SetLastAccessTime(svFileCRC, IO.File.GetLastAccessTime(szFileName))
            IO.File.SetLastWriteTime(svFileCRC, IO.File.GetLastWriteTime(szFileName))

        End If

        'Ajout ProgressBar CRC
        FormBox.ToolStripStatusLabelFM.Text = Nothing

    End Function

    Public Function AppendString(ByVal svData As String) As String
        'Declaration des variables
        Dim lCrc32Value As Integer
        Dim FileData() As Byte
        Dim FileSize As Integer
        Dim nvData As Integer = 1

        FileSize = Len(svData)

        ReDim FileData(FileSize - 1)

        While nvData < FileSize
            System.Windows.Forms.Application.DoEvents()
            FileData(nvData - 1) = CByte(Asc(Mid(svData, nvData, 2)))
            nvData += 2
        End While

        'Initialise le tableau du CRC
        lCrc32Value = InitCrc32()
        'Calcule le CRC
        lCrc32Value = AddCrc32(FileData, lCrc32Value)
        'Ajouste le CRC par XOR
        lCrc32Value = GetCrc32(lCrc32Value)

        'Info
        AppendString = CStr(Hex(lCrc32Value))

    End Function
End Module